src/PIL/ImageFile.py

```diff
@@ -670,6 +650,32 @@ def setimage(self, im, extents=None):
         ):
             raise ValueError("Tile cannot extend outside image")
 
+
+class PyDecoder(PyCodec):
```

Comment:
Do these non-overridden methods needs testing?

Reply:
Ok, there should now be specific tests, and I've also implemented a PyEncoder for BLP to test it in the general sense.